docs/src/prototypes/formValidation/FormValidationOnChange.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react'
+import { Form, Button, Input, RadioGroup } from '@stardust-ui/react'
+import Formsy from 'formsy-react'
+import FormsyFormField from './FormsyFormField'
+
+class FormValidationOnChange extends React.Component<any, any> {
+  state = { buttonDisabled: false }
+
+  render() {
+    return (
+      <Form
+        as={Formsy}
+        onSubmit={() => {
+          alert('Form submitted')
+        }}
+        onValid={() => this.setState({ buttonDisabled: false })}
+        onInvalid={() => this.setState({ buttonDisabled: true })}
+        fields={this.getFields()}
+      />
+    )
+  }
+
+  getFields = () => {
+    const { buttonDisabled } = this.state
+    const fields = [
+      {
+        control: { as: Input },
+        name: 'full-name-on-change',
+        id: 'full-name-on-change',
+        key: 'full-name',
+        required: true,
+        label: 'Full Name',
+        validationErrors: {
```

Comment:
Is there duplication between prop required and validation rule IsDefaultRequired?  Or should there be warning if required is true and isDefaultRequired not set?  - Not sure these are good ideas - but wanted to think about it.

Reply:
isDefaultRequired inside the validationErrors props is just the error message that should be shown, and the required is the actual rule.